OneSignalSDK/unittest/src/test/java/com/onesignal/ShadowOSUtils.java

```diff
@@ -1,23 +1,63 @@
 package com.onesignal;
 
+import android.content.Context;
+
 import org.robolectric.annotation.Implements;
 
+import java.util.HashSet;
+
 @Implements(OSUtils.class)
 public class ShadowOSUtils {
-   static int deviceType = 1;
-   public static String carrierName = "test1";
 
-   public static int subscribableStatus = 1;
+   private static String FIRE_OS_CLASS = "com.amazon.device.messaging.ADM";
+
+   static int deviceType;
+   public static String carrierName;
+   public static int subscribableStatus;
+
+   /**
+    * Used to mock the existence of FireOS ADM class for Amazon devices
+    * TODO: This could apply to other classes that might not be available in our UnitTests but
+    *  we want to mock in a way that makes sense and tests our SDK the closest to production functionality
+    */
+   private static HashSet<String> availableClasses;
+
+   public static void mockAmazonDevice() {
+      ShadowOSUtils.addClass(ShadowOSUtils.FIRE_OS_CLASS);
+   }
+
+   private static void addClass(String className) {
+      availableClasses.add(className);
+   }
+
+   /**
+    * Reset all static values (should be called before each test)
+    */
+   public static void resetStatics() {
+      deviceType = 0;
+      carrierName = "test1";
+      subscribableStatus = 1;
+
+      availableClasses = new HashSet<>();
+   }
 
    public int getDeviceType() {
+      // Class only available on the FireOS, if this class exists we want to mock Amazon deviceType
```

Comment:
This looks good for now, as `getDeviceType` gets more complex with more platforms we will probably want to not mock this method. But that can be done in the task where we add the new platform.

Reply:
Yeah we would need a better way to mock a class existing within our unit tests and that way when the REAL getDeviceType method is called it just uses that functionality rathe than my `HashSet`